### ISSUE TITLE
Accuracy change with BetterTransformer

### ISSUE THREAD

**Issue by kapilsingh93** [huggingface/optimum#1769]
When transforming the model into BetterTransformer model I'm seeing accuracy drop on the models. 
The output scores changes considerably (upto 1-2 decimal points of precision). 
**Is accuracy change expected when switching to BetterTransformer ?** I'm not performing any ORT compilation or quantization on the model. 
From what I know FlashAttention is not supposed to change any accuracy since it is an exact attention score algorithm. Hence I'm not sure what is causing this change in score. 

Steps to reproduce
```
from transformers import AutoModelForSequenceClassification , AutoTokenizer
from optimum.bettertransformer import BetterTransformer

tokenizer=AutoTokenizer.from_pretrained("BAAI/bge-reranker-large")

original_model = AutoModelForSequenceClassification.from_pretrained("BAAI/bge-reranker-large").to('cuda:0')
transformed_model = BetterTransformer.transform(original_model, keep_original_model=True).to('cuda:0')

sentences_batch=[['do you like fox cookies', 'fox big brown fox']] 
inputs = tokenizer(sentences_batch,padding=True,truncation=True,return_tensors="pt",max_length=512,).to('cuda:0')

better_transformer_scores = transformed_model(**inputs, return_dict=True).logits.view(-1).float()
print(f"BetterTransfomer output: {better_transformer_scores.detach().cpu().numpy().tolist()}")

vanilla_model_scores = original_model(**inputs, return_dict=True).logits.view(-1).float()
print(f"Vanilla model output :{vanilla_model_scores.detach().cpu().numpy().tolist()}")
```
Output
```
BetterTransfomer output: [-7.378745079040527]
Vanilla model output :[-7.3596720695495605]
```
##### System state:
* Package version:
   * transformers == 4.39.1
   * optimum == 1.17.1
   * torch == 2.2.1
* Instance Type : AWS p3.2xlarge ( GPU V100) . (Tied it on A100 as well )
* CUDA Version: 12.2
* GPU Driver Version: 535.104.12

**Comment by fxmarty** [comment]
Hi @kapilsingh93, thank you, I can reproduce, this is not expected, sorry for the issue. Let me fix shortly.

**Comment by fxmarty** [comment]
@kapilsingh93 Interestingly downgrading to torch 2.0.1 fixes the issue... It may be a torch regression

**Comment by fxmarty** [comment]
@kapilsingh93 I would suggest you to downgrade to torch==2.0.1 and see if it fixes your problem. This is weird.

**Comment by kapilsingh93** [comment]
Thanks for the prompt response. I can see that torch 2.0.1 helps in reducing the variance. 
However I still see decent variance in results when comparing the models in fp16 mode ( by using `model.half()` ) .

**Comment by fxmarty** [comment]
@kapilsingh93 Can you share a reproduction in fp16?

**Comment by kapilsingh93** [comment]
Adding `model.half()` to the above code 

```
from transformers import AutoModelForSequenceClassification , AutoTokenizer
from optimum.bettertransformer import BetterTransformer

original_model.half()
transformed_model.half()

tokenizer=AutoTokenizer.from_pretrained("BAAI/bge-reranker-large")

original_model = AutoModelForSequenceClassification.from_pretrained("BAAI/bge-reranker-large").to('cuda:0')
transformed_model = BetterTransformer.transform(original_model, keep_original_model=True).to('cuda:0')

sentences_batch=[['do you like fox cookies', 'fox big brown fox']] 
inputs = tokenizer(sentences_batch,padding=True,truncation=True,return_tensors="pt",max_length=512,).to('cuda:0')

better_transformer_scores = transformed_model(**inputs, return_dict=True).logits.view(-1).float()
print(f"BetterTransfomer output: {better_transformer_scores.detach().cpu().numpy().tolist()}")

vanilla_model_scores = original_model(**inputs, return_dict=True).logits.view(-1).float()
print(f"Vanilla model output :{vanilla_model_scores.detach().cpu().numpy().tolist()}")
```

produces output
```
BetterTransfomer output: [-7.35546875]
Vanilla model output :[-7.3515625]
```
